packages/react-reconciler/src/ReactFiberWorkLoop.new.js

```diff
@@ -412,28 +415,57 @@ export function requestUpdateLane(
           : NoLanes;
     }
 
-    lane = findTransitionLane(
+    return findTransitionLane(
       transitionLanePriority,
       currentEventWipLanes,
       currentEventPendingLanes,
     );
-  } else {
-    // TODO: If we're not inside `runWithPriority`, this returns the priority
-    // of the currently running task. That's probably not what we want.
-    const schedulerPriority = getCurrentPriorityLevel();
+  }
+
+  // TODO: Remove this dependency on the Scheduler priority.
+  // To do that, we're replacing it with an update lane priority.
+  const schedulerPriority = getCurrentPriorityLevel();
+
+  if (decoupleUpdatePriorityFromScheduler) {
+    // In the new strategy, we will track the current update lane priority
+    // inside React and use that priority to select a lane for this update.
+    // For now, we're just logging when they're different so we can assess.
+    const updateLaneSchedulerPriority = lanePriorityToSchedulerPriority(
+      getCurrentUpdateLanePriority(),
+    );
```

Comment:
I think we're still on different pages regarding what this warning is supposed to do.

- `findUpdateLane` returns an update lane.
- In the existing algorithm, it reads the current Scheduler priority and converts it to a LanePriority.
- In the future algorithm, it will instead read different priority field that is tracked by React. This field is already a LanePriority, so no conversion is needed.
- The warning should fire if the two algorithms produce different results — i.e., if they result in different lane priorities.

So the warning should compare the LanePriority for the update, not the Scheduler priority.

Reply:
That was my understanding as well, I think we just differed on the implementing details and what gets logged.